packages/analytics-browser/test/browser-client.test.ts

```diff
@@ -261,6 +261,163 @@ describe('browser-client', () => {
+        }).promise;
+        const diagnosticTrack = jest.spyOn(client.config.diagnosticProvider as Diagnostic, 'track');
+        client.config.transportProvider = transportProvider;
+        await client.track('event_type', { userId: 'user_0' }).promise;
+
+        expect(diagnosticTrack).toHaveBeenCalledTimes(0);
+      });
+
+      test.each([null, new Error()])('should diagnostic track when 0 unexpected error', async (res) => {
+        const transportProvider = {
+          send: jest.fn().mockImplementationOnce(() => {
+            return Promise.resolve(res);
+          }),
+        };
+
+        await client.init(apiKey, {
+          defaultTracking: false,
+        }).promise;
+        const diagnosticTrack = jest.spyOn(client.config.diagnosticProvider as Diagnostic, 'track');
+        client.config.transportProvider = transportProvider;
+        await client.track('event_type', { userId: 'user_0' }).promise;
+
+        expect(diagnosticTrack).toHaveBeenCalledTimes(1);
+        expect(diagnosticTrack).toHaveBeenCalledWith(1, 0, core.UNEXPECTED_DIAGNOSTIC_MESSAGE);
+      });
+
+      test.each([
+        ['api_key', undefined, core.INVALID_OR_MISSING_FIELDS_DIAGNOSTIC_MESSAGE],
+        [undefined, { time: [0] }, core.EVENT_ERROR_DIAGNOSTIC_MESSAGE],
+      ])(
+        'should diagnostic track when 400 invalid response',
+        async (missingField, eventsWithInvalidFields, message) => {
+          const transportProvider = {
+            send: jest.fn().mockImplementationOnce(() => {
+              return Promise.resolve({
+                status: Status.Invalid,
+                statusCode: 400,
+                body: {
+                  error: 'error',
+                  missingField: missingField,
+                  eventsWithInvalidFields: eventsWithInvalidFields,
+                  eventsWithMissingFields: {},
+                  eventsWithInvalidIdLengths: {},
+                  silencedEvents: [],
+                },
+              });
+            }),
+          };
+
+          await client.init(apiKey, {
+            defaultTracking: false,
+          }).promise;
+          const diagnosticTrack = jest.spyOn(client.config.diagnosticProvider as Diagnostic, 'track');
+          client.config.transportProvider = transportProvider;
+          await client.track('event_type', { userId: 'user_0' }).promise;
+
+          expect(diagnosticTrack).toHaveBeenCalledTimes(1);
+          expect(diagnosticTrack).toHaveBeenCalledWith(1, 400, message);
+        },
+      );
+
+      test('should diagnostic track when 429 rate limit when flush', async () => {
+        const transportProvider = {
+          send: jest.fn().mockImplementationOnce(() => {
+            return Promise.resolve({
+              status: Status.RateLimit,
+              statusCode: 429,
+              body: {
+                exceededDailyQuotaUsers: { user_0: 1 },
+                exceededDailyQuotaDevices: {},
+                throttledEvents: [],
```

Comment:
Just to confirm, does the diagnostic endpoint return the same response payloads as HTTPv2 API?

Reply:
Is there any concern with it? Afaik, the success response is similar but not the same. 

diagnostic endpoint: 
```
{
    "code": 200,
    "events_ingested": 1
}
```

HTTT V2 API:
```
{
  "code": 200,
  "events_ingested": 50,
  "payload_size_bytes": 50,
  "server_upload_time": 1396381378123
}
```

Reply:
Sorry, reviewing to quick. This makes sense.